File: plugins/superpowers-plus/skills/performance-audit/test-fixtures/cpp-sample/service.cpp
#include "shop.hpp"

#include <iostream>

namespace shop {

// Enriches an order's line items by looking each one up by id, logging progress.
std::vector<Item> handle_order(const Db& db, const std::vector<std::string>& ids) {
    std::vector<Item> items;
    for (const auto& id : ids) {
        Item it = db.query_item(id);
        std::cout << "loaded item " << it.id << std::endl;
        items.push_back(it);
    }
    return items;
}

// Computes the order aggregates (revenue, tax, shipping).
Totals compute_totals(const Db& db, const std::string& order_id) {
    Totals t;
    t.revenue = db.aggregate_revenue(order_id);
    t.tax = db.aggregate_tax(order_id);
    t.shipping = db.aggregate_shipping(order_id);
    return t;
}

}  // namespace shop

File: plugins/superpowers-plus/skills/performance-audit/test-fixtures/go-sample/service.go
// Package shop implements a small HTTP order service.
//
// Eval fixture for the performance-audit skill (illustrative). Answer key in
// expected-findings.md — assessor-only; do not read it when auditing this fixture.
package shop

import (
	"database/sql"
	"encoding/json"
	"net/http"
)

type Server struct {
	db *sql.DB
}

// HandleOrder enriches an order's line items and returns them.
func (s *Server) HandleOrder(w http.ResponseWriter, r *http.Request) {
	ids := r.URL.Query()["item"]

	var items []Item
	for _, id := range ids {
		row := s.db.QueryRow("SELECT id, name, price FROM items WHERE id = $1", id)
		var it Item
		if err := row.Scan(&it.ID, &it.Name, &it.Price); err == nil {
			items = append(items, it)
		}
	}

	client := &http.Client{}
	resp, _ := client.Get("http://pricing/quote?order=" + r.URL.Query().Get("order"))
	var quote Quote
	json.NewDecoder(resp.Body).Decode(&quote)

	json.NewEncoder(w).Encode(map[string]any{"items": items, "quote": quote})
}

// Totals fetches the revenue, tax, and shipping aggregates for an order.
func (s *Server) Totals(orderID string) (Totals, error) {
	revenue, err := s.fetchRevenue(orderID)
	if err != nil {
		return Totals{}, err
	}
	tax, err := s.fetchTax(orderID)
	if err != nil {
		return Totals{}, err
	}
	ship, err := s.fetchShipping(orderID)
	if err != nil {
		return Totals{}, err
	}
	return Totals{revenue, tax, ship}, nil
}
